### PR TITLE
Remove .NET Core install templates from pipelines

### PR DESCRIPTION
**Pull request by damonbarry**
I believe we can finally remove the .NET Core install templates from our pipelines because either:
1. The pipelines are running on 1ES-hosted agents, which already have the latest version of .NET Core 2.1 and 3.1 installed
2. The pipelines are running on MS-hosted agents, which [already have recent version of .NET Core 2.1 and 3.1 installed](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-Readme.md#net-core-sdk)

This change removes the templates and their references. The templates were installing a hard-coded, much older version of the SDK, and therefore missed the latest security patches and other fixes. I also updated the Functions binding project to replace some deprecated properties that were causing errors when built against the latest .NET Core SDK.

Note that this PR does not remove the template that configures the default SDK on Linux.

I tested the three pipelines impacted by this change: dotnet checkin, dotnet CI, and images release.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [X] Title of the pull request is clear and informative.
- [X] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [X] concise summary of tests added/modified
	- [X] local testing done.

### CODE REVIEW

File: edge-modules/functions/binding/src/Microsoft.Azure.WebJobs.Extensions.EdgeHub/Microsoft.Azure.WebJobs.Extensions.EdgeHub.csproj [hunk 1]
```diff
@@ -18,14 +18,20 @@
...
     <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288890</PackageIconUrl>
+    <PackageIcon>icon.png</PackageIcon>
     <PackageProjectUrl>https://github.com/Azure/iotedge</PackageProjectUrl>
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
     <PackageTags>IoT Microsoft Azure IoTEdge Functions</PackageTags>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Include="..\..\..\..\..\LICENSE" Pack="true" PackagePath=""/>
+    <None Include="images\icon.png" Pack="true" PackagePath=""/>
```
Comment (and-rewsmith) [inline edge-modules/functions/binding/src/Microsoft.Azure.WebJobs.Extensions.EdgeHub/Microsoft.Azure.WebJobs.Extensions.EdgeHub.csproj:32]: What's the image for and where will it be displayed?

**Review by and-rewsmith — APPROVED** [review]
Looks good. Recommend:
1. Making sure all instances of install-dotnet*.yaml are no longer present
2. Running pipelines to confirm.

**Comment by and-rewsmith** [comment]
When I recommended running the pipelines, I was rly only concerned with images release build. But I see that you preserved the dotnet configuration script to allow switching between versions. Your call on what we should run.

**Comment by damonbarry** [comment]
> Looks good. Recommend:
> 
> 1. Making sure all instances of install-dotnet*.yaml are no longer present
> 2. Running pipelines to confirm.

Yep, I confirmed that all instances are removed, and that all impacted pipelines pass with the changes.

File: edge-modules/functions/binding/src/Microsoft.Azure.WebJobs.Extensions.EdgeHub/Microsoft.Azure.WebJobs.Extensions.EdgeHub.csproj (hunk 1 above)
Comment (damonbarry) [reply edge-modules/functions/binding/src/Microsoft.Azure.WebJobs.Extensions.EdgeHub/Microsoft.Azure.WebJobs.Extensions.EdgeHub.csproj:32]: I'm pretty sure it's displayed in nuget.org search results, not sure if it's displayed anywhere else. I'm a little annoyed at having to check in a png file, especially in such a seemingly trivial case, but it preserves existing behavior and it's very minor, so the easiest path.

**Pull request merged**